controllers/watcherapi_controller.go

```diff
@@ -639,6 +717,25 @@ func (r *WatcherAPIReconciler) reconcileDelete(ctx context.Context, instance *wa
 			}
 		}
 	}
+	// remove finalizers to cert secrets if present
+	if instance.Spec.TLS.CaBundleSecretName != "" {
+		result, err := r.removeFinalizerFromSecret(ctx, helper, instance, instance.Spec.TLS.CaBundleSecretName)
+		if (result != ctrl.Result{} || err != nil) {
+			return result, err
+		}
+	}
+	if instance.Spec.TLS.API.Internal.SecretName != nil && *instance.Spec.TLS.API.Internal.SecretName != "" {
+		result, err := r.removeFinalizerFromSecret(ctx, helper, instance, *instance.Spec.TLS.API.Internal.SecretName)
```

Comment:
Thinking again about the finalizer, there is a corner case here in case a user create a watcher with TLS and later removes the TLS.API.Internal.SecretName or move to a different one. Then, the finalizer will not be removed from the first.

I don't know if there is a way to manage that tbh, we'd need to have access to the previous state of the CR during the reconcile loop, wich i'm not sure if it's available.

We could fix the case, assume it will need manual action to delete the finalizer on that case or don't add the finalizer. wdyt?

Reply:
good catch, I had not though about that. After a quick search I did not see any way to access old state, it seems to go against the general ideas of kubernetes. I think it might be best not to add the finalizers, since it's trying to solve an unlikely problem

Reply:
+1